mithril-common/src/cardano_transactions_preloader.rs

```diff
@@ -14,6 +15,32 @@ use crate::signable_builder::TransactionsImporter;
 use crate::signed_entity_type_lock::SignedEntityTypeLock;
 use crate::StdResult;
 
+/// CardanoTransactionsPreloaderActivationState
+#[async_trait]
+pub trait CardanoTransactionsPreloaderActivationState: Send + Sync {
+    /// Determine if the Cardano Transactions Preloader should preload.
+    async fn is_activated(&self) -> StdResult<bool>;
+}
+
+/// CardanoTransactionsPreloaderActivation
+pub struct CardanoTransactionsPreloaderActivation {
```

Comment:
This type should probably be defined in the aggregator as it is the implementation that it will use.

Reply:
We believe that this minimalist implementation has the right place here to facilitate its basic use. Every context that need it can simply use it, for instance in the testing contexts in the Signer node.

Reply:
It'd be better to stick to the usual way of doing it: implementing Fake/Mock for testing and keeping node specific implementations where they belong.